contracts/src/op/OPSuccinctDisputeGame.sol

```diff
@@ -0,0 +1,362 @@
+    /// @return The creator of the dispute game.
+    function gameCreator() public pure returns (address) {
+        return _getArgAddress(0x00);
+    }
+
+    /// @notice Getter for the root claim.
+    /// @dev `clones-with-immutable-args` argument #2
+    /// @return The root claim of the DisputeGame.
+    function rootClaim() public pure returns (Claim) {
+        return Claim.wrap(_getArgBytes32(0x14));
+    }
+
+    /// @notice Getter for the parent hash of the L1 block when the dispute game was created.
+    /// @dev `clones-with-immutable-args` argument #3
+    /// @return The parent hash of the L1 block when the dispute game was created.
+    function l1Head() public pure returns (Hash) {
+        return Hash.wrap(_getArgBytes32(0x34));
+    }
+
+    /// @notice Getter for the extra data.
+    /// @dev `clones-with-immutable-args` argument #4
+    /// @return extraData_ Any extra data supplied to the dispute game contract by the creator.
+    function extraData() public pure returns (bytes memory extraData_) {
+        // The extra data starts at the second word within the cwia calldata and
+        // is 32 bytes long.
+        extraData_ = _getArgBytes(0x54, 0x20);
+    }
+
+    function challengeWithProof(
+        bytes32 aggregationVkey,
+        address verifier,
+        bytes32 _outputRoot,
+        uint256 _l2BlockNumber,
+        bytes32 rollupConfigHash,
+        bytes32 rangeVkeyCommitment,
+        bytes memory _proof
+    ) external {
+        AggregationOutputs memory publicValues = AggregationOutputs({
+            l1Head: l1Head().raw(),
+            l2PreRoot: startingOutputRoot.root.raw(),
+            claimRoot: _outputRoot,
+            claimBlockNum: _l2BlockNumber,
+            rollupConfigHash: rollupConfigHash,
+            rangeVkeyCommitment: rangeVkeyCommitment
+        });
+
+        ISP1Verifier(verifier).verifyProof(
+            aggregationVkey,
+            abi.encode(publicValues),
+            _proof
+        );
+
+        claimData.counteredBy = msg.sender;
+    }
+
+    /// @notice The l2BlockNumber of the disputed output root in the `OPSuccinctDisputeGame`.
+    function l2BlockNumber() public pure returns (uint256 l2BlockNumber_) {
+        l2BlockNumber_ = _getArgUint256(0x54);
+    }
+
+    /// @notice If all necessary information has been gathered, this function should mark the game
+    ///         status as either `CHALLENGER_WINS` or `DEFENDER_WINS` and return the status of
+    ///         the resolved game. It is at this stage that the bonds should be awarded to the
+    ///         necessary parties.
+    /// @dev May only be called if the `status` is `IN_PROGRESS`.
+    /// @return status_ The status of the game after resolution.
+    function resolve() external returns (GameStatus status_) {
+        // INVARIANT: Resolution cannot occur unless the game is currently in progress.
+        if (status != GameStatus.IN_PROGRESS) revert GameNotInProgress();
+
```

Comment:
Shouldn't there be another check here to ensure that the challenge clock hasn't expired yet? Otherwise, the prover could finalize immediately?